scripts/pages/leaderboards/leaderboards.ts

```diff
@@ -152,16 +173,73 @@ class LeaderboardsHandler {
 	/**
 	 * Hide the button to go to the end of run page.
 	 */
-	onMapLoad(_isOfficial: boolean) {
+	onMapLoad(isOfficial: boolean) {
 		this.panels.endOfRunButton.visible = false;
+
+		if (isOfficial) {
+			return; // Load official leaderboard tracks instead
+		}
+
+		// Try to load tracks from local zones
+		const mapZoneData = MomentumTimerAPI.GetActiveZoneDefs();
+		if (mapZoneData) {
+			// Main track
+			{
+				const trackStr = $.Localize('#Leaderboards_Tracks_Main');
+				const item = $.CreatePanel('Label', this.panels.tracksDropdown, trackStr, {
```

Comment:
The thing about inlining `text: $.Localize('#Leaderboards_Tracks_Main')` was just because it doesn't seem helpful to define `trackStr` separate - variable name doesn't convey much information so don't think there's any point declaring it separately. But that's just a nit, not a big deal, okay as-is.